### PR TITLE
Create servicemonitor.yaml

### PR DESCRIPTION
**Pull request by aravind-bluescape**
Creating service monitor for vault

| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | NA
| License         | Apache 2.0

### What's in this PR?

This PR is raised to include a new 'service monitoring' feature for vault. A ServiceMonitor is used to define an application you wish to scrape metrics from within Kubernetes, the controller will action the ServiceMonitors we define and automatically build the required Prometheus configuration. Currently we do not have a service monitor for vault due to which we do not receive any metric related to vault

### Why?

We can get vault related metrics post implementing a service monitor

### Checklist

- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)

### CODE REVIEW

**Comment by bonifaido** [comment]
Please bump the chart version @aravind-bluescape

**Comment by aravind-bluescape** [comment]
> Please bump the chart version @aravind-bluescape

Chart version bumped. Could you please review and merge?

File: charts/vault-operator/values.yaml [hunk 1]
```diff
@@ -80,3 +80,10 @@ serviceAccount:
   # The name of the service account to use.
   # If not set and create is true, a name is generated using the fullname template
   name: ""
+monitoring:
+  # Create a Vault Operator ServiceMonitor object
+  serviceMonitor:
+    enabled: false
+    additionalLabels: {}
+    metricRelabelings: []
+    relabelings: []
```
Comment (bonifaido) [inline charts/vault-operator/values.yaml:89]: We are having some lint issue here with trailing spaces after the last line.

Comment (aravind-bluescape) [reply charts/vault-operator/values.yaml]: Removed trailing spaces and tested - local linting was successful

**Review by bonifaido — APPROVED** [review]

**Pull request merged**